### ISSUE TITLE
Phone/Email validation

### ISSUE THREAD

**Issue by ZavenArra** [Greenstand/treetracker-android#820]
validate the phone number
validate the email as non-empty and not containing spaces (but not necessarily an email)

We only want one of these, if a user switches from one to the other, only store the most recently entered.

**Comment by EngrTaofeek** [comment]
@ZavenArra currently for the user entity there is phone number, email and wallet field. The wallet field is extracted from either which is available. I can modify the wallet field to take the most recent entry between the two. Would that be enough or the phone and email should be removed from the User's entity?

**Comment by ZavenArra** [comment]
wallet should take most recent entry
clear out the least recent entry (leave phone set if phone is most recent, but clear in email - or vice versa)

**Comment by Elforama** [comment]
@EngrTaofeek are you sure? I didn't change the validation logic for email. It might still require an email (but it shouldn't)

**Comment by EngrTaofeek** [comment]
@Elforama you are right. It's the description of the issue that confused me. @ZavenArra if it's not necessary an email there is no need for an @ symbol any string should do. It would be weird to require an @ symbol and not a domain (.com etc.)

**Comment by ZavenArra** [comment]
I think that a minimum we should require an @ symbol.  Most users will enter a full domain, but there are some special cases that we can support by allowing only an @ symbol.
